### PR TITLE
Check syntax of each solution

### PR DESCRIPTION
**Pull request by Krishn1412**
Solves #20

### CODE REVIEW

File: src/gpt_resolve/pdf_generator.py [hunk 1]
```diff
@@ -59,6 +86,22 @@ def generate_solutions_pdf(
...
+    # Validate each solution before adding to the document
+    for sol_file in solution_files:
+        print(f"Validating solution: {sol_file.name}")
+        content = sol_file.read_text(encoding="utf-8")
+        
+        if not compile_solution(content):
+            print(f"Error: Solution '{sol_file.name}' has LaTeX syntax errors. Please fix it.")
+            return 
+
+    print("All solutions are valid. Generating the final PDF...")
```
Comment (lgabs) [inline src/gpt_resolve/pdf_generator.py]: Nice! But I think there is a more concise way to solve this:
- instead of looping through questions twice, you can use the existent loop through solution_files to, for each content read, try to compile it with compile_solution in a tmp file as you've suggested
- if the compilation is successful, continue the loop as you've done

Besides, with this implementation the user will only see the first error found. It'd be better if you collect wrong files in a list so that we print all files with errors at the end of the loop.

**Review by lgabs — CHANGES_REQUESTED** [review]

File: src/gpt_resolve/pdf_generator.py [hunk 2]
```diff
@@ -3,6 +3,33 @@
...
+        doc.generate_tex(str(tex_file))
+        try:
+            result = subprocess.run(
+                ["pdflatex", "-output-directory", temp_dir, tex_file],
+                capture_output=True, text=True
+            )
+            return result.returncode == 0
+        except subprocess.CalledProcessError as e:
+            print(f"Error compiling LaTeX content: {e}")
+            return False
```
Comment (lgabs) [inline src/gpt_resolve/pdf_generator.py]: About this function, I have two suggestions:
- a more clear name like validate_latex_content since it returns a bool, with docstring like "Validate a LaTeX document by attempting to compile it, checking for syntax validity."
- subprocess does not seem necessary, as you can use `doc.generate_pdf(tex_file)`, right?

**Comment by lgabs** [comment]
Also, do you mind writing some unit tests? Currently the generate_solutions_pdf function has only a test for empty dir and another for a latex string which is correct. With you implementation, we could test for incorrect latex string examples.

**Comment by Krishn1412** [comment]
I have made some some changes based on your comments. Can you take a look again? I'll work on the unit tests.

File: src/gpt_resolve/pdf_generator.py [hunk 3]
```diff
@@ -59,10 +83,22 @@ def generate_solutions_pdf(
...
+    sol_files_with_issues = []
     # Add each solution to document
     for sol_file in solution_files:
+        print(f"Validating solution: {sol_file.name}")
         content = sol_file.read_text(encoding="utf-8")
+
+        if not validate_latex_content(content):
+            print(f"Error: Solution '{sol_file.name}' has LaTeX syntax errors. Please fix it.")
+            sol_files_with_issues.append(sol_file)
+            continue
```
Comment (lgabs) [inline src/gpt_resolve/pdf_generator.py]: Nice! The only thing left here is that any errors will count in `sol_files_with_issues` and loop will continue, but in the end the code will try to compile the whole pdf again, generating the same error that currently happens (for many questions, the traceback will be big).

So if you print errors only outside the loop in case of `sol_files_with_issues` not empty, it'd be solved:
```
for sol_file in solution_files:
    ...
    if not validate_latex_content(content):
        sol_files_with_issues.append(sol_file)
        continue

if sol_files_with_issues:
    raise ValueError("Errors in one or more solutions:\n' + '\n'.join(sol_files_with_issues) + '\n\nPlease check for syntax errors.")
    
...
```

**Review by lgabs — CHANGES_REQUESTED** [review]

**Comment by lgabs** [comment]
@Krishn1412 do you have any updates here?

**Comment by Krishn1412** [comment]
Sorry about the delay @lgabs, I had some place to be. Can you take a look now?

**Comment by lgabs** [comment]
No problem @Krishn1412 , thanks a lot! Looks great now for merging. Do you have the time to add some unit test for invalid case? If so, I'll wait for your updates; if not, I'll create another issue to add this unit test afterwards.

**Comment by Krishn1412** [comment]
Sure @lgabs  I'll add some unit tests. Do you want me to add them in the same PR or create a new one?

**Comment by lgabs** [comment]
You can do it in different PR, no problem.
